server/pulp/server/db/model/__init__.py

```diff
@@ -631,7 +631,7 @@ def unit_key_str(self):
         """
         The unit key represented as a string ordered by unit key fields alphabetically
         """
-        return str(sorted([getattr(self, key) for key in self.unit_key_fields]))
+        return str(sorted(self.unit_key.items()))
```

Comment:
Actually, on further consideration, I don't think it's possible to create a 100% unique (to the unit key) unit key string using string concatenation.  Submitting a *hashing* solution instead that is certain to be both deterministic and unique.